File: AEDS2/Verde/TP1/PalindromoRecursivo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>
#include <stdlib.h>

bool isPalindromo(char string[], int index) {
    bool resp = true;
    if(index < strlen(string)) {
        if(string[index] != string[strlen(string) - 1 - index]) {
            resp = false;
        }
        else {
            index++;
            resp = isPalindromo(string, index);
        }
    }
    return resp;
}

int main() {
    char *string = (char*)malloc( 500 * sizeof(char));
    int index = 0;
    while(strcmp(string, "FIM") != 0) {
        scanf(" %[^\n]", string);
        if(isPalindromo(string, index)) {
            printf("SIM\n");
        }
        else if(strcmp(string, "FIM") != 0) {
            printf("NAO\n");
        }
    }
    return 0;
}

File: AEDS2/Verde/TP1/PalindromoRecursivo.java
public class PalindromoRecursivo {
    // Método para verificar se duas strings são iguais.
    public static boolean equalStrings(String str_1, String str_2) {
        if(str_1.length() != str_2.length()) {
            return false;
        }

        for(int i = 0; i < str_1.length(); i++) {
            if(str_1.charAt(i) != str_2.charAt(i)) {
                return false;
            }
        }

        return true;
    }


    // Método para verificar se uma string é um palindromo
    public static boolean isPalindromo(String string, int index) {
        boolean resp = true;

        if(index < string.length()) {
            if(string.charAt(index) != string.charAt(string.length() - 1 - index)) {
                resp = false;
            }
            else {
                index++; 
                resp = isPalindromo(string, index); // Chamada recursiva com o próprio indice
            }
        }
        return resp;

    }

    // Main
    public static void main(String[] args) {
        String string;
        do {
            string = MyIO.readLine();
            if(!equalStrings(string, "FIM")) {
                int index = 0;
                if(isPalindromo(string, index)) {
                    MyIO.println("SIM");
                }
                else {
                    MyIO.println("NAO");
                }
            }
        }while(!equalStrings(string, "FIM"));
    }
}
